app/uk/gov/hmrc/cataloguefrontend/healthindicators/HealthIndicatorsController.scala

```diff
@@ -48,18 +49,26 @@ class HealthIndicatorsController @Inject() (
       } yield result
     }
 
-  def indicatorsForRepoType(repoType: String, repoName: String): Action[AnyContent] =
+  def indicatorsForRepoType(): Action[AnyContent] =
     Action.async { implicit request =>
-      RepoType
-        .parse(repoType)
+      form
+        .bindFromRequest()
         .fold(
-          _ => Future.successful(Redirect(routes.HealthIndicatorsController.indicatorsForRepoType(RepoType.Service.asString, repoName))),
-          r =>
+          formWithErrors => Future.successful(BadRequest(HealthIndicatorsLeaderBoard(Seq.empty, Seq.empty, Seq.empty, formWithErrors))),
+          validForm =>
             for {
-              indicatorsWithTeams <- healthIndicatorsService.findIndicatorsWithTeams(r)
-            } yield Ok(HealthIndicatorsLeaderBoard(indicatorsWithTeams, r, repoName, RepoType.values))
+              indicatorsWithTeams <- healthIndicatorsService.findIndicatorsWithTeams(validForm.repoType.getOrElse(RepoType.Service))
+              teams <- teamsAndRepositoriesConnector.allTeams
+              indicators = indicatorsFilteredByTeam(indicatorsWithTeams, validForm.team)
+            } yield Ok(HealthIndicatorsLeaderBoard(indicators, RepoType.values, teams.sortBy(_.name.asString), form.fill(validForm)))
```

Comment:
`TeamName` already has an implicit ordering. You can just call `.sort`.
Maybe order in `allTeams` function for all clients?

Reply:
This is the `Team` object (not `TeamName`) so I can either add implicit ordering by name or cut this down to just use `.sortBy(_.name)`?